common/types/src/monitoring.rs

```diff
@@ -0,0 +1,122 @@
+use std::{collections::HashSet, sync::LazyLock, time::SystemTime};
+
+use nym_crypto::asymmetric::identity::{PrivateKey, PublicKey, Signature};
+use nym_mixnet_contract_common::MixId;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+static NETWORK_MONITORS: LazyLock<HashSet<String>> = LazyLock::new(|| {
+    let mut nm = HashSet::new();
+    nm.insert("5VsPyLbsBCq9PAMWmjKkToteVAKNabNqex6QwDf5fWzt".to_string());
+    nm
+});
+
+#[derive(Debug, Serialize, Deserialize, JsonSchema, Clone)]
+pub struct MixnodeResult {
+    pub mix_id: MixId,
+    pub identity: String,
+    pub owner: String,
+    pub reliability: u8,
+}
+
+impl MixnodeResult {
+    pub fn new(mix_id: MixId, identity: String, owner: String, reliability: u8) -> Self {
+        MixnodeResult {
+            mix_id,
+            identity,
+            owner,
+            reliability,
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Clone)]
+pub struct GatewayResult {
+    pub identity: String,
+    pub owner: String,
+    pub reliability: u8,
+}
+
+impl GatewayResult {
+    pub fn new(identity: String, owner: String, reliability: u8) -> Self {
+        GatewayResult {
+            identity,
+            owner,
+            reliability,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, JsonSchema)]
+#[serde(untagged)]
+pub enum MonitorResults {
+    Mixnode(Vec<MixnodeResult>),
+    Gateway(Vec<GatewayResult>),
+}
+
+#[derive(Serialize, Deserialize, JsonSchema)]
+pub struct MonitorMessage {
+    results: MonitorResults,
+    signature: String,
+    signer: String,
+    timestamp: i64,
+}
+
+impl MonitorMessage {
+    fn message_to_sign(results: &MonitorResults, timestamp: i64) -> Vec<u8> {
+        let mut msg = match serde_json::to_vec(results) {
+            Ok(msg) => msg,
+            Err(_) => return Vec::new(),
```

Comment:
if serialisation fails, you still want to attach the timestamp so that somebody couldn't replay your signature on an empty message